client/python/openlineage/client/transport/http.py

```diff
@@ -70,10 +83,544 @@ def create_token_provider(auth: dict[str, str]) -> TokenProvider:
+                    return
+        else:
+            event_id = hashlib.md5(event_str.encode("utf-8")).hexdigest()
+
+        body, headers = self._prepare_request(event_str)
+
+        self.event_tracker.add_event(event_id)
+
+        # Wait for space in the regular capacity portion of the queue
+        while self.event_queue.qsize() >= self.configured_queue_size:
+            time.sleep(0.01)  # Brief wait before checking again
+
+        # Add to queue - include run_id and event_type for START event tracking
+        self.event_queue.put((event_id, body, headers, run_id, event_type))
+        log.debug(f"Queued event with event_id {event_id} for async emission to {self.config.url}")
+
+    def wait_for_completion(self, timeout: float = 30.0) -> bool:
+        # Block until all events are processed or timeout is reached.
+        log.debug("Waiting for events to be sent in wait_for_completion.")
+        start_time = time.time()
+        while time.time() - start_time < timeout:
+            if self.event_tracker.all_processed():
+                log.debug("All events were processed.")
+                return True
+
+            # Brief sleep before checking again
+            time.sleep(0.01)
+
+        return self.event_tracker.all_processed()
+
+    def get_stats(self) -> dict[str, int]:
+        return self.event_tracker.get_stats()
+
+    def shutdown(self, wait: bool = True, timeout: float = 30.0) -> bool:
+        if wait:
+            result = self.wait_for_completion(timeout)
+            self.should_exit.set()
+            if self.worker_thread.is_alive():
+                self.worker_thread.join(timeout=1.0)
+            return result
+        self.should_exit.set()
+        return True
+
+
+class SyncEmitter(Emitter):
+    def __init__(self, config: HttpConfig) -> None:
+        self.config = config
+        self.url = config.url
+        self.endpoint = config.endpoint
+        self.timeout = config.timeout
+        self.verify = config.verify
+        self.compression = config.compression
+
+    def emit(self, event: Event) -> httpx.Response:
+        # If anyone overrides debuglevel manually, we can potentially leak secrets to logs.
+        # Override this setting to make sure it does not happen.
+        prev_debuglevel = http_client.HTTPConnection.debuglevel
+        http_client.HTTPConnection.debuglevel = 0
+
+        event_str = Serde.to_json(event)
+
+        body, headers = self._prepare_request(event_str)
+
+        transport = httpx.HTTPTransport(retries=self.config.retry.get("total", 5))
+
+        with httpx.Client(
+            timeout=self.config.timeout, verify=self.config.verify, follow_redirects=True, transport=transport
+        ) as client:
+            resp = client.post(
+                url=urljoin(self.url, self.endpoint),
+                content=body,
+                headers=headers,
+            )
 
-    return Session()
+        http_client.HTTPConnection.debuglevel = prev_debuglevel
```

Comment:
This is probably should be set within try..finally block, otherwise raising an exception will left current debuglevel instead of resetting it